crates/dyn-abi/src/ext/event.rs

```diff
@@ -0,0 +1,301 @@
+use crate::{DynSolValue, DynToken, Error, ResolveSolType, Result};
+use alloc::vec::Vec;
+use alloy_json_abi::Event;
+use alloy_primitives::{Log, B256};
+use alloy_sol_types::Decoder;
+
+mod sealed {
+    pub trait Sealed {}
+    impl Sealed for alloy_json_abi::Event {}
+}
+use sealed::Sealed;
+
+/// Provides event encoding and decoding for the [`Event`] type.
+///
+/// This trait is sealed and cannot be implemented for types outside of this
+/// crate. It is implemented only for [`Event`].
```

Comment:
Should this be a `Resolve` like interface that outputs a `DynEvent` and then uses that to (en/de)code?

Reply:
The only thing it would do is resolve the types, I don't think so

Reply:
i guess this is a question of how we conceptualize dyn-abi. if it has the same scope as sol-types then we should have dyn equivalents to everything in sol-types

the original idea was to have a runtime representation that was format-independent, and then be able to convert JSON or solidity lines, or type strings to the runtime representation. we're now kinda mixing that, where _some_ things are resolved to the runtime repr, and some aren't. which leads to a somewhat weird API

Reply:
or for a concrete thing, if we do this as an ext trait then we duplicate code across these two impls
```
impl EventExt for Event {}
impl EventExt for syn_solidity::ItemEvent {}
```

I would prefer something like

```
trait ResolveEvent {
    fn resolve(&self) -> DynEvent
}

impl ResolveEvent for JsonEvent {}
impl ResolveEvent for syn_soldity::ItemEvent {}

```